v3io/dataplane/transport/httpclient.py

```diff
@@ -165,27 +219,41 @@ def _send_request_on_connection(self, request, connection):
                 )
                 if retries_left == 0:
                     raise
+
                 retries_left -= 1
-                connection.close()
+
+                # Close failed connection
+                with contextlib.suppress(Exception):
+                    current_connection.close()
+                # Create a new connection for retry, thread-safe
+                with self._lock:
+                    if self._closed:
+                        raise RuntimeError("Transport closed during request retry") from e
+                    current_connection = self._create_connection(self._host, self._ssl_context)
```

Comment:
Here too, it doesn't seem like creating the connection belongs in a locked section.

Reply:
See above